receipt/src/main/java/com/hyperwallet/android/receipt/view/ListReceiptFragment.java

```diff
@@ -0,0 +1,307 @@
+        registerObservers();
+    }
+
+    private void registerObservers() {
+        mListReceiptViewModel.getReceiptList().observe(getViewLifecycleOwner(), new Observer<PagedList<Receipt>>() {
+            @Override
+            public void onChanged(PagedList<Receipt> transferMethods) {
+                mListReceiptAdapter.submitList(transferMethods);
+            }
+        });
+
+        mListReceiptViewModel.getReceiptErrors().observe(getViewLifecycleOwner(),
+                new Observer<Event<List<HyperwalletError>>>() {
+            @Override
+            public void onChanged(Event<List<HyperwalletError>> listEvent) {
+                if (!listEvent.isContentConsumed()) {
+                    mOnLoadReceiptErrorCallback.showErrorOnLoadReceipt(listEvent.getContent());
+                }
+            }
+        });
+
+        mListReceiptViewModel.isLoadingData().observe(getViewLifecycleOwner(), new Observer<Boolean>() {
+            @Override
+            public void onChanged(Boolean loading) {
+                if (loading) {
+                    mProgressBar.setVisibility(View.VISIBLE);
+                } else {
+                    mProgressBar.setVisibility(View.GONE);
+                }
+            }
+        });
+    }
+
+    void retry() {
+        mListReceiptViewModel.retryLoadReceipts();
+    }
+
+    interface OnLoadReceiptErrorCallback {
+
+        void showErrorOnLoadReceipt(@NonNull final List<HyperwalletError> errors);
+    }
+
+    private static class ListReceiptItemDiffCallback extends DiffUtil.ItemCallback<Receipt> {
+
+        @Override
+        public boolean areItemsTheSame(@NonNull Receipt oldItem, @NonNull Receipt newItem) {
+            return oldItem.hashCode() == newItem.hashCode()
+                    && Objects.equals(oldItem, newItem);
+        }
+
+        @Override
+        public boolean areContentsTheSame(@NonNull Receipt oldItem, @NonNull Receipt newItem) {
+            return oldItem.hashCode() == newItem.hashCode()
+                    && Objects.equals(oldItem, newItem);
+        }
+    }
+
+    private static class ListReceiptAdapter
+            extends PagedListAdapter<Receipt, ListReceiptAdapter.ReceiptViewHolder> {
+
+        static final int HEADER_VIEW_TYPE = 1;
+        static final int DATA_VIEW_TYPE = 0;
+
+        ListReceiptAdapter(@NonNull final DiffUtil.ItemCallback<Receipt> diffCallback) {
+            super(diffCallback);
+        }
+
+        @Override
+        public int getItemViewType(int position) {
+            if (position != 0) {
+                Receipt previous = getItem(position - 1);
+                Receipt current = getItem(position);
+                if (isDataViewType(previous, current)) {
+                    return DATA_VIEW_TYPE;
+                }
+            }
+            return HEADER_VIEW_TYPE;
+        }
+
+        boolean isDataViewType(@NonNull final Receipt previous, @NonNull final Receipt current) {
```

Comment:
@peter-joseph Maybe we shouldn't do it in the adapter?

Reply:
Its ok to do this here we are not saving it, it will be GC'd after we don't use it right away, but we cannot save date instance to Receipt as well since createdOn in Receipt object is GMT, and in presentation layer we need to change it on whatever locale is the mobile user using so no.. this is good enough